### PR TITLE
Generators

### PR DESCRIPTION
**Pull request by msisaifu**
I try to translate the Generators chapter. It's my first translation. Let me know if anything wrong.

Thanks :)

### CODE REVIEW

File: 1-js/12-generators-iterators/1-generators/article.md [hunk 1]
```diff
@@ -380,27 +383,27 @@ alert( generator.next(4).value ); // "3 * 3 = ?"
...
 ## generator.throw
 
-As we observed in the examples above, the outer code may pass a value into the generator, as the result of `yield`.
+উপরের কোড হতে আমরা বুঝতে পারি, আমরা জেনারেটরে একটি ভ্যালু পাঠাতে পারি, যা `yield` এর রেজাল্ট হয়।
 
-...But it can also initiate (throw) an error there. That's natural, as an error is a kind of result.
+...কিন্ত আমরা চাইলে সেখানে একটি এরর (থ্রো) করতে পারি। এটি স্বাভাবিক কেননা ইরোরও একধরনের রেজাল্ট।
 
-To pass an error into a `yield`, we should call `generator.throw(err)`. In that case, the `err` is thrown in the line with that `yield`.
+`yield` এ এরর পাস করতে চাইলে আমাদের `generator.throw(err)` কে কল করা লাগবে। এর ফলে, the `err` is thrown in the line with that `yield`.
```
Comment (ahamed) [inline 1-js/12-generators-iterators/1-generators/article.md:404]:
```suggestion
`yield` এ এরর পাস করতে চাইলে আমাদের `generator.throw(err)` কে কল করা লাগবে। এই ক্ষেত্রে, `err` টি যে লাইনে `yield` আছে সেখানে পাঠানো হবে।
```

**Comment by msisaifu** [comment]
> Here, with some typoes, I've given some suggestions. You can approve of them.

thanks, approved :)

File: 1-js/12-generators-iterators/1-generators/01-pseudo-random-generator/solution.md [hunk 2]
```diff
@@ -35,4 +35,4 @@ alert(generator()); // 282475249
 alert(generator()); // 1622650073
 ```
 
-That also works. But then we lose ability to iterate with `for..of` and to use generator composition, that may be useful elsewhere.
+এটিও কাজ করে। কিন্তু এ জন্য  জেনারেট কম্পোজিশনে আমরা `for..of` এর মাধ্যমে ইটারেট করতে পারব না, এটি অন্য কোথাও ব্যবহারযোগ্য হতে পারে।
```
Comment (jamal-pb95) [inline 1-js/12-generators-iterators/1-generators/01-pseudo-random-generator/solution.md:38]:
```suggestion
এটিও কাজ করে। কিন্তু এ জন্য  জেনারেটর কম্পোজিশনে আমরা `for..of` এর মাধ্যমে ইটারেট করতে পারব না, এটি অন্য কোথাও ব্যবহারযোগ্য হতে পারে।
```

File: 1-js/12-generators-iterators/1-generators/01-pseudo-random-generator/task.md [hunk 3]
```diff
@@ -1,29 +1,30 @@
...
+এমন অনেক ক্ষেত্র আছে যেখানে আমাদের র‍্যান্ডম ডাটা জেনারেট করা লাগে।
 
-One of them is testing. We may need random data: text, numbers, etc. to test things out well.
+এর মধ্যে একটি হল টেস্টিং। টেস্ট করার জন্য আমাদের প্রয়োজন হতে পারে র‍্যান্ডম ডাটাঃ টেক্সট, নাম্বার ইত্যাদি।
 
-In JavaScript, we could use `Math.random()`. But if something goes wrong, we'd like to be able to repeat the test, using exactly the same data.
+জাভাস্ক্রিপ্টে আমরা `Math.random()` ব্যবহার করতে পারি। কিন্ত যদি আমরা কোন ভুল করে ফেলি এবং টেস্ট আবার একই ডাটা দিয়ে রিপিট করতে চাই। 
 
-For that, so called "seeded pseudo-random generators" are used. They take a "seed", the first value, and then generate the next ones using a formula so that the same seed yields the same sequence, and hence the whole flow is easily reproducible. We only need to remember the seed to repeat it.
+এজন্য তথাকথিত "সীডেড সুডো-র‍্যান্ডম জেনারেটর" ব্যবহার করা হয়। এইগুলো প্রথম ভ্যালুতে একটি "সীড" নেয়, এবং তারপর একটি সূত্র ব্যবহার করে পরবর্তিটি জেনারেট করে যাতে করে একই সীড একই ক্রমধারা তৈরি করতে পারে, আর এতে করে পুরো প্রবাহটি খুব সহজে পুনরায় গঠন করা যায়। এর জন্য আমাদের কেবল কোন সীডটি পুনরায় করতে হবে সেটা মনে রাখলেই চলবে।
```
Comment (jamal-pb95) [inline 1-js/12-generators-iterators/1-generators/01-pseudo-random-generator/task.md:10]:
```suggestion
এজন্য তথাকথিত "সীডেড সুডো-র‍্যান্ডম জেনারেটর" ব্যবহার করা হয়। এইগুলো প্রথম ভ্যালুতে একটি "সীড" নেয়, এবং তারপর একটি সূত্র ব্যবহার করে পরবর্তীটি জেনারেট করে যাতে করে একই সীড একই ক্রমধারা অনুযায়ী তৈরি করতে পারে, আর এতে করে পুরো প্রবাহটি খুব সহজে পুনরায় গঠন করা যায়। এর জন্য আমাদের কেবল কোন সীডটি পুনরায় করতে হবে সেটা মনে রাখলেই চলবে।
```

File: 1-js/12-generators-iterators/1-generators/article.md [hunk 4]
```diff
@@ -1,14 +1,14 @@
-# Generators
+# জেনারেটর
 
-Regular functions return only one, single value (or nothing).
+সাধারণ ফাংশনগুলো শুধুমাত্র একটি মান রিটার্ন করে (অথবা কিছুই রিটার্ন করে না)। 
 
-Generators can return ("yield") multiple values, one after another, on-demand. They work great with [iterables](info:iterable), allowing to create data streams with ease.
+জেনারেটরের মাধ্যমে  আমরা চাহিদা অনুযায়ী একটির পর আরেকটি এভাবে অনেক মান রিটার্ন ("yield") করতে পারি। এরা [ইটারেবলের](info:iterable) সাথে দারুণ কাজ করে, এর মাধ্যমে সহজে ডাটা স্ট্রীম তৈরী করা যায়।
```
Comment (jamal-pb95) [inline 1-js/12-generators-iterators/1-generators/article.md:5]:
```suggestion
জেনারেটরের মাধ্যমে আমরা চাহিদা অনুযায়ী একটির পর আরেকটি এভাবে অনেক মান রিটার্ন ("yield") করতে পারি। এরা [ইটারেবলের](info:iterable) সাথে দারুণ কাজ করে, এর মাধ্যমে সহজে ডাটা স্ট্রীম তৈরী করা যায়।
```

File: 1-js/12-generators-iterators/1-generators/article.md [hunk 5]
```diff
@@ -18,9 +18,11 @@ function* generateSequence() {
 }
 ```
 
-Generator functions behave differently from regular ones. When such function is called, it doesn't run its code. Instead it returns a special object, called "generator object", to manage the execution.
+জেনারেটর ফাংশন এর বিহেভিয়ার সাধারণ ফাংশন গুলো থেকে ভিন্ন।
```
Comment (jamal-pb95) [inline 1-js/12-generators-iterators/1-generators/article.md:21]:
```suggestion
জেনারেটর ফাংশন এর আচরণ সাধারণ ফাংশন গুলো থেকে ভিন্ন।
```

File: 1-js/12-generators-iterators/1-generators/article.md [hunk 6]
```diff
@@ -18,9 +18,11 @@ function* generateSequence() {
...
+যখন এই ধরণের ফাংশন গুলো কল হয়, এরা এদের কোড রান করে না।
```
Comment (jamal-pb95) [inline 1-js/12-generators-iterators/1-generators/article.md:22]:
```suggestion
যখন এই ধরণের ফাংশন গুলো কল করা হয়, তখন এরা এদের কোড রান করে না।
```

File: 1-js/12-generators-iterators/1-generators/article.md [hunk 7]
```diff
@@ -29,24 +31,24 @@ function* generateSequence() {
...
 
-The result of `next()` is always an object with two properties:
-- `value`: the yielded value.
-- `done`: `true` if the function code has finished, otherwise `false`.
+`next()` হল একটি অবজেক্ট যার দুটি প্রোপার্টি আছে:
+- `value`: ইয়েল্ডেড মানটি।
+- `done`: `true` যদি ফাংশনের কোড শেষ হয়, অন্যথায় `false`।
 
-For instance, here we create the generator and get its first yielded value:
+উদাহরণস্বরুপ, এখানে আমরা জেনারেটর তৈরি করেছি এবং আমরা এটির প্রথম ইয়েল্ডেড মানটি পাইঃ
```
Comment (jamal-pb95) [inline 1-js/12-generators-iterators/1-generators/article.md:51]:
```suggestion
উদাহরণস্বরুপ, এখানে আমরা জেনারেটর তৈরি করেছি এবং আমরা এটির প্রথম ইয়েল্ডেড মানটি পাওয়া যায়ঃ
```

File: 1-js/12-generators-iterators/1-generators/article.md [hunk 8]
```diff
@@ -154,30 +159,30 @@ let sequence = [0, ...generateSequence()];
 alert(sequence); // 0, 1, 2, 3
 ```
 
-In the code above, `...generateSequence()` turns the iterable generator object into an array of items (read more about the spread syntax in the chapter [](info:rest-parameters-spread#spread-syntax))
+উপরের কোডে, `...generateSequence()` ইটারেবল জেনারেটর অবজেক্টটি অ্যারেতে রুপান্তরিত হয় (স্প্রেড অপারেটর সম্পর্কে আরো জানতে এটি পড়ুন [](info:rest-parameters-spread-operator#spread-operator))
 
-## Using generators for iterables
+## ইটারেবল এর জন্য জেনারেটরের ব্যবহার
```
Comment (jamal-pb95) [inline 1-js/12-generators-iterators/1-generators/article.md:164]:
```suggestion
## ইটারেবলের জন্য জেনারেটরের ব্যবহার
```

**Comment by msisaifu** [comment]
done

**Pull request merged**